src/img_boss/make_vdi.py

```diff
@@ -0,0 +1,145 @@
+                f.image.image_url, raw)
+
+        if not raw.endswith(".raw"):
+            try:
+                orig=raw
+                # Try to uncompress
+                if raw.endswith(".gz"):
+                    print "gunzip %s\n" % raw
+                    subprocess.check_output(["gunzip", raw])
+                    raw = raw[0:-3]
+                if raw.endswith(".tar.bz2"):
+                    print "tar xv -C %s -f %s\n" % (os.path.dirname(raw), raw)
+                    new_raw = subprocess.check_output(["tar", "xv",
+                                                       "-C", os.path.dirname(raw),
+                                                       "-f", raw]
+                                                       ).rstrip('\n')
+                    subprocess.check_output(["rm", raw])
+                    raw = raw[0:-8] + ".raw"
+                    os.rename(os.path.join(os.path.dirname(raw), new_raw),
+                              raw)
+#                    new_raw = subprocess.check_output(["tar", "xv",
+#                                                       "-C", "/tmp",
+#                                                       "-f", raw]
+#                                                       ).rstrip('\n')
+#                    os.unlink(raw)
+#                    raw = raw[0:-8] + ".raw"
+#                    shutil.copy(os.path.join("/tmp", new_raw),
+#                                raw)
+#                    os.unlink(os.path.join("/tmp", new_raw))
+
+                if raw.endswith(".bz2"):
+                    print "bunzip2 %s\n" % raw                    
+                    subprocess.check_output(["bunzip2", raw])
+                    raw = raw[0:-4]
+            except subprocess.CalledProcessError,e:
+                return False, "Failed whilst trying to uncompress %s.\n%s\n" % (raw, e.output)
+            if not raw.endswith(".raw") and not os.path.isfile(raw):
+                return False, "Failed to convert %s to a .raw called %s" % (orig, raw)
+            print "Got %s" % raw
+        try:
+            vdi = raw[0:-3]+"VDI"
+            command = [ "VBoxManage", "convertfromraw",
+                        raw, vdi]
+            if f.vboxmanage and f.vboxmanage.format:
+                command.append("--format=%s" % f.vboxmanage.format)
+            else:
+                command.append("--format=%s" % "VDI")
+
+            if f.vboxmanage and f.vboxmanage.variant:
+                command.append("--variant=%s" % f.vboxmanage.variant)
+
+            print " ".join(command)
+            subprocess.check_output(command)
+            subprocess.check_call(["chmod", "0664", vdi])
+            subprocess.check_call(["bzip2", "--fast", vdi])
+            subprocess.check_call(["chown", "-R", "img", os.path.dirname(raw)])
+        except subprocess.CalledProcessError,e:
+            return False, "Running convertfromraw failed:\n%s" % e.output
+
+        # Cleanup
+        os.remove(raw)
```

Comment:
If the  raw image is removed, should the image_url, image_type and others that refer to it be updated it the WI?

Reply:
that makes sense

Reply:
I was thinking more like changing the image_type to 'vdi' and image_url to point to the vdi file